hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/compact/CompactNullablePrimitiveInteroperabilityTest.java

```diff
@@ -135,6 +130,59 @@ public void testWritePrimitiveReadNullable() {
         assertReadAsNullable(serializedRecord);
     }
 
+    private <T> SerializationService createSerializationService(Supplier<CompactSerializer<T>> serializerSupplier) {
+        CompactSerializationConfig compactSerializationConfig = new CompactSerializationConfig();
+        compactSerializationConfig.addSerializer(serializerSupplier.get());
+        return new DefaultSerializationServiceBuilder()
+                .setSchemaService(schemaService)
+                .setConfig(new SerializationConfig().setCompactSerializationConfig(compactSerializationConfig))
+                .build();
+    }
+
+    @Test
+    public void testWritePrimitiveReadNullableCustomSerializer() {
+        FixedSizeFieldsDTO fixedSizeFieldsDTO = createFixedSizeFieldsDTO();
+        SerializationService serializationService = createSerializationService(FixedSizeFieldsDTOSerializerReadingNullable::new);
+
+        Data data = serializationService.toData(fixedSizeFieldsDTO);
+        FixedSizeFieldsDTO obj = serializationService.toObject(data);
+
+        assertEquals(obj, fixedSizeFieldsDTO);
+
+        ArrayOfFixedSizeFieldsDTO arrayOfFixedSizeFieldsDTO = createArrayOfFixedSizeFieldsDTO();
+        CompactSerializationConfig compactSerializationConfig2 = new CompactSerializationConfig();
+        compactSerializationConfig2.addSerializer(new ArrayOfFixedSizeFieldsDTOSerializerReadingNullable());
+        SerializationService serializationService2 = new DefaultSerializationServiceBuilder()
+                .setSchemaService(schemaService)
+                .setConfig(new SerializationConfig().setCompactSerializationConfig(compactSerializationConfig2))
+                .build();
+
+        Data data2 = serializationService2.toData(arrayOfFixedSizeFieldsDTO);
+        ArrayOfFixedSizeFieldsDTO obj2 = serializationService2.toObject(data2);
+
+        assertEquals(obj2, arrayOfFixedSizeFieldsDTO);
```

Comment:
Feels like it would be better to have this as a separate test. I believe it is not a good practice to catch two birds with a one stone in unit tests